test/doc_comments_test.dart

```diff
@@ -0,0 +1,199 @@
+// Copyright 2018 Google Inc. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'package:sass/src/ast/sass.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('documentation comments', () {
+    group('SCSS syntax:', () {
+      test('variable declarations are found', () {
+        final contents = r'''
+            /// Results my vary.
+            $vary: 5.16em;''';
+        final stylesheet = Stylesheet.parseScss(contents);
+        final variable = stylesheet.children
+                .firstWhere((child) => child is VariableDeclaration)
+            as VariableDeclaration;
+
+        expect(variable.comment.docComment, equals('Results my vary.'));
+      });
+
+      test('function rules are found', () {
+        final contents = r'''
+            /// A fun function!
+            @function fun($val) {
+              // Not a doc comment.
+              @return ($val / 1000) * 1em;
+            }''';
+        final stylesheet = Stylesheet.parseScss(contents);
+        final function = stylesheet.children
+            .firstWhere((child) => child is FunctionRule) as FunctionRule;
+
+        expect(function.comment.docComment, equals('A fun function!'));
+      });
+
+      test('mixin rules are found', () {
+        final contents = r'''
+            /// Mysterious mixin.
+            @mixin mystery {
+              // All black.
+              color: black;
+              background-color: black;
+            }''';
+        final stylesheet = Stylesheet.parseScss(contents);
+        final mix = stylesheet.children
+            .firstWhere((child) => child is MixinRule) as MixinRule;
+
+        expect(mix.comment.docComment, equals('Mysterious mixin.'));
+      });
+
+      test('attached to the correct children', () {
```

Comment:
It's usually good practice to have each test case cover exactly one concept. For this one, I'd split out "double-slash comments aren't parsed as doc comments" and "doc comments aren't carried over across members" into two separate cases.

Reply:
Done.